src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -1,79 +1,244 @@
+      bucketResponse <- createBucket(googleProject, clusterName, bucketName)
+      // Add initialization files after the bucket has been created
+      storageObjectsResponse <- initializeBucketObjects(googleProject, clusterName, bucketName)
+    } yield {
+      (bucketResponse, storageObjectsResponse)
+    }
+  }
+
+  /* Create a bucket in the given google project for the initialization files when creating a cluster */
+  private def createBucket(googleProject: GoogleProject, clusterName: String, bucketName:String): Future[Bucket] = {
+    Future {
+      val bucket = new Bucket().setName(bucketName)
+      val bucketInserter = storage.buckets().insert(googleProject, bucket)
+      try {
+        executeGoogleRequest(bucketInserter) // returns a Bucket
       } catch {
-        case e: GoogleJsonResponseException => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"Google Request Failed: ${e.getDetails.getMessage}"))
+        case e: GoogleJsonResponseException => throw new BucketNotCreatedException(googleProject, clusterName)
       }
     }
   }
 
+  /* Process the templated cluster init script and put all initialization files in the init bucket */
+  private def initializeBucketObjects(googleProject: GoogleProject, clusterName: String, bucketName: String): Future[Array[StorageObject]] = {
+    Future {
+      // Get the init script
+      val initScriptRaw = scala.io.Source.fromFile(dataprocConfig.configFolderPath + dataprocConfig.initActionsScriptName).mkString
+      // Get a map of the template values to be replaced
+      val replacements = ClusterInitValues(googleProject, clusterName, bucketName, dataprocConfig).toJson.asJsObject.fields
+      // Replace templated values in the raw init script with actual values before populating the bucket
+      val initScript = replacements.foldLeft(initScriptRaw)((a, b) => a.replaceAllLiterally("$(" + b._1 + ")", b._2.toString()))
+      // Create InputStreamContent using the new init script
+      val content = new InputStreamContent(null, new ByteArrayInputStream(initScript.getBytes(StandardCharsets.UTF_8)))
+      // Put init script in bucket
+      val initScriptStorageObject = populateBucket(googleProject, bucketName, dataprocConfig.initActionsScriptName, content)
+
+      // Create an array of all the certs, cluster docker compose file, and the site.conf for the apache proxy
+      val certs = Array(dataprocConfig.jupyterServerCrtName, dataprocConfig.jupyterServerKeyName, dataprocConfig.jupyterRootCaPemName,
+        dataprocConfig.clusterDockerComposeName, dataprocConfig.jupyterProxySiteConfName)
+      // Put the rest of the initialization files in the init bucket
+      val storageObjects = certs.map { certName => populateBucket(googleProject, bucketName, certName, new FileContent(null, new File(dataprocConfig.configFolderPath, certName))) }
+      // Return an array of all the Storage Objects
+      storageObjects :+ initScriptStorageObject
+    }
```

Comment:
Needs more error handling. The file I/O stuff can potentially throw `IOException`, which we should catch and wrap in a `LeoException`.